pkg/utils/helmwrapper/helm_wrapper.go

```diff
@@ -0,0 +1,454 @@
+package helmwrapper
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"k8s.io/klog"
+	kpath "k8s.io/utils/path"
+	"kubesphere.io/kubesphere/pkg/utils/idutils"
+	"os"
+	"os/exec"
+	"path"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+var (
+	UninstallNotFoundFormat = "Error: uninstall: Release not loaded: %s: release: not found"
+	StatusNotFoundFormat    = "Error: release: not found"
+)
+
+type HelmRes struct {
+	Message string
+}
+
+type releaseStatus struct {
+	Name string `json:"name,omitempty"`
+	Info *Info  `json:"info,omitempty"`
+}
+
+// copy from helm
+// Info describes release information.
+type Info struct {
+	// FirstDeployed is when the release was first deployed.
+	FirstDeployed time.Time `json:"first_deployed,omitempty"`
+	// LastDeployed is when the release was last deployed.
+	LastDeployed time.Time `json:"last_deployed,omitempty"`
+	// Deleted tracks when this object was deleted.
+	Deleted time.Time `json:"deleted"`
+	// Description is human-friendly "log entry" about this release.
+	Description string `json:"description,omitempty"`
+	// Status is the current state of the release
+	Status string `json:"status,omitempty"`
+	// Contains the rendered templates/NOTES.txt if available
+	Notes string `json:"notes,omitempty"`
+}
+
+type helmRlsStatus struct {
+	Name       string `json:"name"`
+	Namespace  string `json:"namespace"`
+	Revision   int    `json:"revision"`
+	Status     string `json:"status"`
+	Chart      string `json:"chart"`
+	AppVersion string `json:"app_version"`
+}
+
+var _ helmI = &helmWrapper{}
```

Comment:
I see, you're using `helm` command to do the upgrade/uninstall. Are there APIs from helm we can use instead of using `heml` directly?

Reply:
There are a lot of tricks in helm's go package, and many behaviors are inconsistent with the helm command after using it directly, which is a lesson we learned from the legacy code of openpitrix.
So in the end we concluded that using the helm command directly is the most efficient way